astropy/units/physical.py

```diff
@@ -485,19 +485,16 @@ def def_physical_type(unit: core.UnitBase, name: str | set[str]) -> None:
     unit_already_in_use = physical_type_id in _physical_unit_mapping
     if unit_already_in_use:
         physical_type = _physical_unit_mapping[physical_type_id]
-        physical_type_names |= set(physical_type)
-        physical_type.__init__(unit, physical_type_names)
```

Comment:
Mypy reported: `Accessing "__init__" on an instance is unsound, since instance.__init__ could be from an incompatible subclass`.

Reply:
I think your change is indeed for the better. Alternatively, `PhysicalType` should gain an update method or so, but we can do that if there is more need -- inside the module, using a private attribute would seem fine.

Reply:
> I think your change is indeed for the better.
Agreed

> Alternatively, PhysicalType should gain an update method

I would advocate for making PT immutable and slotted for safety and speed, especially if we parametrize Q with it.